### PR TITLE
Add FA derivers for Strelka, Varscan (& mock for Virmid & Mutect)

### PR DESCRIPTION
**Pull request by ihodes**
Also reorganized the display code to be more programmatic.

### CODE REVIEW

File: derive_frequency_of_alleles.py [hunk 1]
```diff
@@ -1,13 +1,71 @@
...
+    """Returns the deriver for a given variant caller. The deriver is a function
+    which takes a VCF record and outputs a VCF record which has the FA attribute
+    set.
+
+    Caller must be one of 'strelka', 'somaticsniper', 'virmid', 'varscan',
+    'mutect'.
+    """
+    caller = caller.lower()
+
+    if caller == 'strelka':
```
Comment (danvk) [inline derive_frequency_of_alleles.py]: just use a Dict for this? Referencing un-set keys in a Dict is an exception in python, so it's just as safe.

File: derive_frequency_of_alleles.py [hunk 2]
```diff
@@ -1,13 +1,71 @@
...
+
+
+def _derive_varscan_fa(record):
+    # Varscan outputs FA already, but in a string formatted as 'XX.XX%' under
+    # the key FREQ, so we just need to pull that out and process it correctly.
+    for sample in record.samples:
+        add_format_field(record, 'FA')
+        freq_string = sample.data.FREQ
+        frequency_of_allele = float(freq_string[:-1]) / 100
+        if frequency_of_allele in (0, 1):
```
Comment (danvk) [inline derive_frequency_of_alleles.py]: comment? this is cryptic.

Comment (danvk) [inline derive_frequency_of_alleles.py]: maybe incorporate this into add_field -- if the value is a float that's really an integer, it shouldn't ever be harmful to do this conversion.

File: derive_frequency_of_alleles.py [hunk 3]
```diff
@@ -1,13 +1,71 @@
+import os
```
Comment (danvk) [inline derive_frequency_of_alleles.py:6]: Add a top level docstring explaining what this does, something like:

"""Add an FA (Frequency of Allelle) field to a VCF file."""

File: derive_frequency_of_alleles.py [hunk 4]
```diff
@@ -55,16 +114,17 @@ def call_data(kvs):
...
-    vcf_in.close()
+    _, caller, infile = sys.argv
+    basename = os.path.splitext(infile)[0]
+    deriver = deriver_for(caller)
+
+    with open(infile) as vcf_in:
+        records = vcf.Reader(vcf_in)
+        records.formats['FA'] = FA_FORMAT
+
+        with open(basename+'.derivedFA.vcf', 'w') as output:
```
Comment (danvk) [inline derive_frequency_of_alleles.py]: this is pretty pedantic, but the python style guide does say to always put spaces around binary operators:
http://google-styleguide.googlecode.com/svn/trunk/pyguide.html?showone=Whitespace#Whitespace

File: display.js [hunk 5]
```diff
@@ -1,25 +1,50 @@
+  ///////////////
+ // Utilities //
+///////////////
+
+function reverseStringTupleComparator(a,b) {
+    a = a.replace(/\(|\)|\s/g, "").split(",").map(function(v){return parseFloat(v);});
+    b = b.replace(/\(|\)|\s/g, "").split(",").map(function(v){return parseFloat(v);});
+    if (a.length !== b.length)  throw TypeError("Tuples must be of the same length.");
+    for (var i in a) {
```
Comment (danvk) [inline display.js]: This isn't the right way to loop over arrays in JS. Just do:

for (var i = 0; i < a.length; i++) {
}

File: display.js [hunk 6]
```diff
@@ -1,25 +1,50 @@
+  ///////////////
+ // Utilities //
+///////////////
+
+function reverseStringTupleComparator(a,b) {
+    a = a.replace(/\(|\)|\s/g, "").split(",").map(function(v){return parseFloat(v);});
```
Comment (danvk) [inline display.js]: 4 spaces --> 2 spaces

Comment (danvk) [inline display.js]: .map(parseFloat) should work, right?

File: display.js [hunk 7]
```diff
@@ -1,25 +1,50 @@
+  ///////////////
+ // Utilities //
+///////////////
+
+function reverseStringTupleComparator(a,b) {
```
Comment (danvk) [inline display.js:5]: add a comment that says what the expected input format is

File: display.js [hunk 8]
```diff
@@ -1,25 +1,50 @@
...
+        }
+    }
+    return 0; // a == b
+}
 _.sum = function(list) { return list.reduce(function(a,b){return a + b;}, 0); };
 
+
+
+  ////////////////////
+ //       ETL      //
```
Comment (danvk) [inline display.js:25]: what is ETL?

Comment (ihodes) [inline display.js:25]: Extract transform load--data munging

Comment (danvk) [inline display.js:25]: Change the comment to that?

**Pull request merged**